parser/parser_expression.go

```diff
@@ -243,12 +246,85 @@ func (p *Parser) parseCreation() node.ExpressionNode {
 	if p.isSymbol(token.OpenParen) {
 		return p.parseStructCreation(abstractNode, identifier)
 	} else if p.isSymbol(token.OpenBracket) {
-		// todo array creation
+		return p.parseArrayCreation(abstractNode, identifier)
 	}
 
 	return p.newErrorNode(fmt.Sprintf("Unsupported creation type with %s", p.currentToken.Literal()))
 }
 
+func (p *Parser) parseArrayInitialization() *node.ArrayInitializationNode {
+	abstractNode := p.newAbstractNode()
+	p.checkAndSkipNewLines(token.OpenBrace) // skip '{'
+	if !p.isEnd() && !p.isSymbol(token.OpenBrace) {
+		var expressions []node.ExpressionNode
+		expressions = append(expressions, p.parseExpression())
+		for !p.isEnd() && !p.isSymbol(token.CloseBrace) {
+			p.checkAndSkipNewLines(token.Comma)
+			expressions = append(expressions, p.parseExpression())
+		}
+		p.checkAndSkipNewLines(token.CloseBrace)
+
+		return &node.ArrayInitializationNode{
+			AbstractNode: abstractNode,
+			Values:       expressions,
+		}
+	}
+
+	arrayInitialization := &node.ArrayInitializationNode{
+		AbstractNode: abstractNode,
+	}
+
+	if !p.isEnd() {
+		var expressions []node.ExpressionNode
+
+		expressions = append(expressions, p.parseArrayInitialization())
+
+		for !p.isEnd() && !p.isSymbol(token.CloseBrace) {
+			p.checkAndSkipNewLines(token.Comma)
+			expressions = append(expressions, p.parseArrayInitialization())
+		}
+
+		p.checkAndSkipNewLines(token.CloseBrace)
+
+		arrayInitialization.Values = expressions
+	} else {
+		p.addError("Invalid array initialization")
+	}
+
+	return arrayInitialization
+}
+
+func (p *Parser) parseArrayCreation(abstractNode node.AbstractNode, identifier string) node.ExpressionNode {
+	arrayType := p.parseTypeWithIdentifier(abstractNode, identifier)
+
+	// Initialization using values
+	if valueCreationNode, ok := arrayType.(*node.ArrayTypeNode); ok {
```

Comment:
You cast `arrayType` to `ArrayTypeNode`. Logically, it is not comprehensible why it has to be like that.

I would refactor it like that:
- first create a type
```
       typeNode := &node.BasicTypeNode{
		AbstractNode: abstractNode,
		Identifier:   identifier,
	}
```
- skip open bracket `[`
- Then create ArrayValueCreationNode if not close bracket

```
        if !p.isSymbol(token.CloseBracket) {
		// create ArrayValueCreationNode
	}
```

With that, you can remove the special case in `parseArrayType`.

Reply:
I think this solution will not work with nested types

Reply:
```
 if !p.isSymbol(token.CloseBracket) {
		// create ArrayValueCreationNode
	}
```

Theres a general mistake in there. ValueCreation is new int[]{1, 2}...

Reply:
@rpfister102 It works like this: https://github.com/bazo-blockchain/lazo/pull/64/commits/e8c898e99f7e1c7e3a005c83e896d0749b8186ae